pandas/core/groupby/generic.py

```diff
@@ -1138,13 +1138,51 @@ def nsmallest(
     def idxmin(
         self, axis: Axis | lib.NoDefault = lib.no_default, skipna: bool = True
     ) -> Series:
+        if axis is lib.no_default:
+            alt = lambda x: Series(x).argmin(skipna=skipna)
+            with com.temp_setattr(self, "observed", True):
+                argmin = self._cython_agg_general("argmin", alt=alt, skipna=skipna)
```

Comment:
There is a bit of an oddity with how missing observations are handled. When there are multiple groupings, we do not include the unobserved categories in e.g. `grouper.result_index` and fill in any unobserved ones in `_wrap_aggregated_output`. However, if there is just a single grouping we _do_ include the unobserved groupings in e.g. `grouper.result_index` and so we don't fill them in later on. This makes this approach not work for some cases of categoricals.

I plan to look into making it so we never include the unobserved categories until `_wrap_aggreagted_output` in the single grouping case. If we can make that work, this approach would work.